iceoryx_hoofs/test/moduletests/test_posix_semaphore_interface.cpp

```diff
@@ -95,9 +160,7 @@ TYPED_TEST(SemaphoreInterfaceTest, PostIncreasesSemaphoreValue)
         ASSERT_FALSE(this->sut->post().has_error());
     }
 
-    auto result = this->sut->getValue();
-    ASSERT_THAT(result.has_error(), Eq(false));
-    EXPECT_THAT(*result, Eq(NUMBER_OF_INCREMENTS));
```

Comment:
Since `isSemaphoreValueEqualTo` uses `post`, the method that is tested here, I would prefer these two checks.